tests/Rector/ClassMethod/DataProviderAnnotationToAttributeRector/Fixture/external-provider.php.inc

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Rector\PHPUnit\Tests\Rector\ClassMethod\DataProviderAnnotationToAttributeRector\Fixture;
+
+use PHPUnit\Framework\TestCase;
+
+final class FooTest extends TestCase
+{
+    /**
+     * @dataProvider Tests\Foo\ExternalProvider::barProvider
+     **/
+    public function testBar()
+    {
+        self::assertTrue(true);
+    }
+}
+
+final class ExternalProvider
+{
+    public static function barProvider()
+    {}
+}
+
+?>
+-----
+<?php
+
+namespace Rector\PHPUnit\Tests\Rector\ClassMethod\DataProviderAnnotationToAttributeRector\Fixture;
+
+use PHPUnit\Framework\TestCase;
+
+final class FooTest extends TestCase
+{
+    #[\PHPUnit\Framework\Attributes\DataProviderExternal(Tests\Foo\ExternalProvider::class, 'barProvider')]
```

Comment:
@samsonasik atm the rector creates
```
#[\PHPUnit\Framework\Attributes\DataProviderExternal('Tests\Foo\ExternalProvider::class', 'barProvider')]
```

but we need a plain class-string without single quotes arround

```
#[\PHPUnit\Framework\Attributes\DataProviderExternal(Tests\Foo\ExternalProvider::class, 'barProvider')]
```

do you have an idea on how to create the annotation arg properly?

Reply:
Using ::class will need FQCN classname with `\` prefix, otherwise, that refer to current namespace, except, short name that already in use statement

Reply:
thx for the hint. my initial question was more about how I could make the rector produce a class-string (which is not sorrounded by single quotes)

Reply:
quoted string should can use String_ node on attribute arg

Reply:
> Using ::class will need FQCN classname with `\` prefix, otherwise, that refer to current namespace, except, short name that already in use statement

I think we need to assume that pre-rector-run existing code either correctly had a use statement or referenced using a FQCN. the rector itself does not change that class

Reply:
if I remember correctly, using annotation `@dataProvider Foo\Class` is same with `@@dataProvider \Foo\Class`, while attribute, it processsed by PHP so : 

```php
DataProviderExternal(Tests\Foo\ExternalProvider::class, 'barProvider')]
```

vs 


```php
DataProviderExternal(\Tests\Foo\ExternalProvider::class, 'barProvider')]
```

are different.

Reply:
ok - added a separate test case for your example